### DISCUSSION TITLE
Add more parameters to the plots

### DISCUSSION THREAD

**Discussion by juanramonua** [shap/shap#3521] [Ideas]
It would be very easy to tailor the plots in more detail by simply adding some extra parameters to the call. Solving these issues now involves reading the code, knowing what kind of objects have been used internally, accessing them with matplotlib and changing them. Which makes the code more difficult and more complicated.

I will give a couple of examples:

**1) plots.scatter()**
- Context: it has a `hist=True` parameter, very useful to know the histogram but internally it has an `alpha=0.1` parameter in the matplotlib hist() function, and the `bins` are autocalculated if a continuous variable (float).
- Problem: With this alpha you can hardly see the histogram with 0.2 or 0.25 you can see it better. As for the `bins` in continuous variables, if they are not displayed properly, they cannot be changed.
- Solution: extra parameters like `plots.scatter( hist_alpha=0.2, hist_bins=8)`.

**2) plots.bar()**
- Context: Text size of the labels on the axes and the values at the end of the horizontal bar.  
- Problem: There are times when the text at the end of the bar looks excessively large.
- Solution: extra parameters like `plots.bar( x_font_size=10, y_font_size=10, bar_font_size=10)`.

**Comment by connortann** [comment]
I agree, this would be great! We have an open issue for this, which I hope can be delivered as part of `0.46.0` :  #3266
